### PR TITLE
New [v6r16] RabbitMQ module 

### PR DESCRIPTION
**Pull request by zmathe**
Created a RabbitMQ module to send and receive messages from queues.

### CODE REVIEW

File: Core/Utilities/RabbitMQConnector.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    try:
+      self.connection = stomp.Connection( [ ( self.host, int( self.port ) ) ], vhost = self.vH )
+      self.connection.start()
+      self.connection.connect( username = self.user, passcode = self.password )
+
+      if self.receiver:
+        self.connection.set_listener( '', self )
+        self.connection.subscribe( destination = '/queue/%s' % self.queueName, id = self.queueName, headers = { 'persistent': 'true' } )
+    except Exception as e:
+      return S_ERROR( 'Failed to setup connection: %s' % e )
```
Comment (fstagni) [inline Core/Utilities/RabbitMQConnector.py]: use new S_ERROR (here and later)

File: Core/Utilities/RabbitMQConnector.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    # Make the actual connection
+    try:
+      self.connection = stomp.Connection( [ ( self.host, int( self.port ) ) ], vhost = self.vH )
+      self.connection.start()
+      self.connection.connect( username = self.user, passcode = self.password )
+
+      if self.receiver:
+        self.connection.set_listener( '', self )
+        self.connection.subscribe( destination = '/queue/%s' % self.queueName, id = self.queueName, headers = { 'persistent': 'true' } )
+    except Exception as e:
```
Comment (fstagni) [inline Core/Utilities/RabbitMQConnector.py]: Can you be more specific?

File: Core/Utilities/MQConnector.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  Abstract class for management of message queue connections
+  Allows to both send and receive messages from a queue
+  """
+
+  def __init__( self ):
+    raise NotImplementedError( 'Cannot instantiate an abstract class' )
+
+  # Callback functions for message receiving mode
+
+  def on_error( self, headers, message ):
```
Comment (chaen) [inline Core/Utilities/MQConnector.py]: you don't use the headers param ?

File: Core/Utilities/MQConnector.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    """
+    Callback function called when an error happens
+    """
+    gLogger.error( message )
+
+  # Rest of functions
+
+  def setQueueParameters( self, system, queueName, parameters ):
+    """
```
Comment (chaen) [inline Core/Utilities/MQConnector.py]: docstring format ?

File: Core/Utilities/MQConnector.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+    # Utility function to lowercase the first letter of a string ( to create valid variable names )
+    toLowerFirst = lambda s: s[:1].lower() + s[1:] if s else ''
+
+    self.queueName = queueName
+
+    # Read and set the parameters
+    for parameter in [ 'Host', 'Port', 'User', 'VH', 'Type' ]:
+      if not parameter in parameters:
+        return S_ERROR( 'The parameter \'%s\' for the queue was not provided' % parameter )
```
Comment (chaen) [inline Core/Utilities/MQConnector.py]: with a set difference you can return the list of missing parameters

File: Core/Utilities/RabbitMQConnector.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+    with self.lock:
+      if self.msgList:
+        msg = self.msgList.pop( 0 )
+      else:
+        return S_ERROR( 'No messages in queue' )
+
+    return S_OK( msg )
+
+  def unsetupConnection( self ):
```
Comment (chaen) [inline Core/Utilities/RabbitMQConnector.py]: unsetup ? How about close like the other class ? :D

**Comment by fstagni** [comment]
IIUC this PR is superseded by https://github.com/DIRACGrid/DIRAC/pull/3054 and can therefore be closed, am I correct?

**Comment by fstagni** [comment]
Closing as per comment above... please comment further if it's not the case.

**Pull request closed without merging**